### PR TITLE
[GPU] Fix issues from unexecuted func tests

### PR DESCRIPTION
**Pull request by byungilm**
### Details:
 - Failed smoke_CLDNN_FP32/StridedSliceLayerTest.Inference/IS=([])_TS={(1.12.100)}_modelType=f32_begin=(0.n6.0)_end=(0.n8.0)_stride=(n1.n2.n1)_begin_m=(1.0.1)_end_m=(1.0.1)_new_axis_m=(0.0.0)_shrink_m=(0.0.0)_ellipsis_m=(0.0.0)_trgDev=GPU
 - *...*

### Tickets:
 - 128235

### CODE REVIEW

**Comment by andrew-k-park** [comment]
Could you check failed TC for StridedSliceLayerTest?

**Review by ahnyoung-paul — APPROVED** [review]

**Comment by isanghao** [comment]
could you run test for this PR?

File: src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp [hunk 1]
```diff
@@ -141,51 +142,53 @@ struct strided_slice_impl : typed_primitive_impl_ocl<strided_slice> {
...
+                if (begin < 0)
+                    begin += out_shape[dim];  // converted value can be negative if the original one was out of bounds
+                if (end < 0)
+                    end += out_shape[dim];
+                bool is_stride_reverse = (stride < 0) ? true : false;
+
+                // Clamping
+                if (should_clamp_begin)
+                    begin = std::min(std::max(begin, (int32_t)0), out_shape[dim]-1);
+                if (should_clamp_end)
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp:178]: I think it is OK to remove this condition. isn't it?

File: src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp [hunk 2]
```diff
@@ -141,51 +142,53 @@ struct strided_slice_impl : typed_primitive_impl_ocl<strided_slice> {
...
+                    end = std::min(std::max(end, (int32_t)0), out_shape[dim]-1);
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp:179]: if `out_shape[dim] = 100`, `begin=0`, `end=100`... this clamping will change `end` to `99`. Then only 99 elements will be copied, while original parameter was to copy 100 elements. Then I think this is not an intended behavior. Is my understanding correct?

Comment (byungilm) [reply src/plugins/intel_gpu/src/graph/impls/ocl/strided_slice.cpp]: Yes, i will update the logic

**Review by isanghao — APPROVED** [review]

**Comment by isanghao** [comment]
no issue from dGPU daily test

**Pull request merged**